code/modules/modular_computers/computers/modular_computer/hardware.dm

```diff
@@ -0,0 +1,153 @@
+// Attempts to install the hardware into apropriate slot.
+/obj/item/modular_computer/proc/try_install_component(var/mob/living/user, var/obj/item/weapon/computer_hardware/H, var/found = 0)
+	// "USB" flash drive.
+	if(istype(H, /obj/item/weapon/computer_hardware/hard_drive/portable))
+		if(enrolled == 1 && !computer_emagged)
+			to_chat(user, "The client management software on this computer rejects \the [portable_drive].")
+			return
+		if(portable_drive)
+			to_chat(user, "This computer's portable drive slot is already occupied by \the [portable_drive].")
+			return
+		found = 1
+		portable_drive = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/hard_drive))
+		if(hard_drive)
+			to_chat(user, "This computer's hard drive slot is already occupied by \the [hard_drive].")
+			return
+		found = 1
+		hard_drive = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/network_card))
+		if(network_card)
+			to_chat(user, "This computer's network card slot is already occupied by \the [network_card].")
+			return
+		found = 1
+		network_card = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/nano_printer))
+		if(nano_printer)
+			to_chat(user, "This computer's nano printer slot is already occupied by \the [nano_printer].")
+			return
+		found = 1
+		nano_printer = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/card_slot))
+		if(card_slot)
+			to_chat(user, "This computer's card slot is already occupied by \the [card_slot].")
+			return
+		found = 1
+		card_slot = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/battery_module))
+		if(battery_module)
+			to_chat(user, "This computer's battery slot is already occupied by \the [battery_module].")
+			return
+		found = 1
+		battery_module = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/processor_unit))
+		if(processor_unit)
+			to_chat(user, "This computer's processor slot is already occupied by \the [processor_unit].")
+			return
+		found = 1
+		processor_unit = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/ai_slot))
+		if(ai_slot)
+			to_chat(user, "This computer's intellicard slot is already occupied by \the [ai_slot].")
+			return
+		found = 1
+		ai_slot = H
+	else if(istype(H, /obj/item/weapon/computer_hardware/tesla_link))
+		if(tesla_link)
+			to_chat(user, "This computer's tesla link slot is already occupied by \the [tesla_link].")
+			return
+		found = 1
+		tesla_link = H
+	if(found)
+		to_chat(user, "You install \the [H] into \the [src]")
+		H.holder2 = src
+		user.drop_from_inventory(H)
+		H.forceMove(src)
+
+// Uninstalls component. Found and Critical vars may be passed by parent types, if they have additional hardware.
+/obj/item/modular_computer/proc/uninstall_component(var/mob/living/user, var/obj/item/weapon/computer_hardware/H, var/found = 0, var/critical = 0)
+	if(portable_drive == H)
+		portable_drive = null
+		found = 1
+	if(hard_drive == H)
```

Comment:
Might be an idea to shove this onto if/else if chains. Should also remove the need for the found variable.

Though, I'm not sure why that's an arg to begin with.